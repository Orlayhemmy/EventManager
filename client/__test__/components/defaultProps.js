export const props = {
  userState: {
    user: {
      isAdmin: true
    },
    isAuth: true
  },
  location: {
    pathname: '/'
  },
  center: {
    status: '',
    message: ''
  }
};

export const center = {
  status: 201,
  message: ''
};

export const bookedEvent = [
  { id: 1, eventTitle: 'Pizza Party', bookedDate: [], isApproved: true },
  { id: 2, eventTitle: 'Wedding Ceremony', bookedDate: [], isApproved: true  },
  { id: 3, eventTitle: 'Seminar', bookedDate: [], isApproved: false  }
];

export const logout = jest.fn();
export const onChange = jest.fn();
