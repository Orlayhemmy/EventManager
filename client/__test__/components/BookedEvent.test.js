import React from 'react';
import EventList from '../../components/CenterDetails/Template/Content/bookedEvents';
import { bookedEvent } from './defaultProps';

describe('Approve Event Component', () => {
  const onClick = jest.fn();
  const eventState = bookedEvent;
  it('should render component without error', () => {
    const wrapper = renderer
      .create(<EventList eventState={eventState} onClick={onClick} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
