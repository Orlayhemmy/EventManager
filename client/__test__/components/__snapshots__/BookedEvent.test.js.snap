// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Approve Event Component should render component without error 1`] = `
<div
  className="col-lg-5"
>
  <div
    className="form-outer text-center d-flex align-items-center"
  >
    <div
      className="form-inner"
    >
      <strong
        className="logo text-primary"
      >
        events scheduled
      </strong>
      <div>
        <table
          cellPadding="0"
          className="table table-responsive table-hover text-center"
        >
          <thead>
            <tr>
              <th>
                title
              </th>
              <th>
                date
              </th>
              <th>
                status
              </th>
              <th />
            </tr>
          </thead>
          <tbody>
            <tr
              id={1}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={1}
                  onClick={[MockFunction]}
                >
                  Pizza Party
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="approve"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-spinner main-color"
                    id={1}
                  />
                </span>
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={1}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={2}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={2}
                  onClick={[MockFunction]}
                >
                  Seminar
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <i
                  className="fa fa-thumbs-up green"
                  id={2}
                />
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={2}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={3}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={3}
                  onClick={[MockFunction]}
                >
                  Pizza Party
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="approve"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-spinner main-color"
                    id={3}
                  />
                </span>
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={3}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={4}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={4}
                  onClick={[MockFunction]}
                >
                  Seminar
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <i
                  className="fa fa-thumbs-up green"
                  id={4}
                />
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={4}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={5}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={5}
                  onClick={[MockFunction]}
                >
                  Pizza Party
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="approve"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-spinner main-color"
                    id={5}
                  />
                </span>
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={5}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={6}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={6}
                  onClick={[MockFunction]}
                >
                  Seminar
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <i
                  className="fa fa-thumbs-up green"
                  id={6}
                />
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={6}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={7}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={7}
                  onClick={[MockFunction]}
                >
                  Pizza Party
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="approve"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-spinner main-color"
                    id={7}
                  />
                </span>
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={7}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={8}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={8}
                  onClick={[MockFunction]}
                >
                  Seminar
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <i
                  className="fa fa-thumbs-up green"
                  id={8}
                />
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={8}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={9}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={9}
                  onClick={[MockFunction]}
                >
                  Pizza Party
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="approve"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-spinner main-color"
                    id={9}
                  />
                </span>
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={9}
                  />
                </span>
              </td>
            </tr>
            <tr
              id={10}
            >
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id={10}
                  onClick={[MockFunction]}
                >
                  Seminar
                </span>
              </td>
              <td>
                22/08/2018
              </td>
              <td>
                <i
                  className="fa fa-thumbs-up green"
                  id={10}
                />
              </td>
              <td>
                <span
                  data-target="#deleteModal"
                  data-toggle="modal"
                  id="trash"
                  onClick={[MockFunction]}
                >
                  <i
                    className="fa fa-trash trash"
                    id={10}
                  />
                </span>
              </td>
            </tr>
            <tr>
              <td>
                <i
                  className="fa fa-spinner main-color"
                />
                <br />
                <span>
                  Pending
                </span>
              </td>
              <td />
              <td />
              <td>
                <i
                  className="fa fa-thumbs-up green"
                />
                <br />
                <span>
                  Approved
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
